x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/IdentifierGenerator.java

```diff
@@ -40,22 +48,27 @@ public static String randomIndexPattern(Feature... features) {
             index.append('.');
         }
         index.append(randomCharacterFrom(validFirstCharacters));
-        for (int i = 0; i < ESTestCase.randomIntBetween(1, 100); i++) {
+        for (int i = 0; i < randomIntBetween(1, 100); i++) {
             index.append(randomCharacterFrom(validCharacters));
         }
         if (canAdd(Features.WILDCARD_PATTERN, features)) {
-            if (ESTestCase.randomBoolean()) {
+            if (randomBoolean()) {
                 index.append('*');
             } else {
-                index.insert(ESTestCase.randomIntBetween(0, index.length() - 1), '*');
+                index.insert(randomIntBetween(0, index.length() - 1), '*');
             }
-        } else if (canAdd(Features.DATE_MATH, features)) {
+        }
+        if (canAdd(Features.DATE_MATH, features)) {
             // https://www.elastic.co/guide/en/elasticsearch/reference/8.17/api-conventions.html#api-date-math-index-names
-            index.insert(0, "<");
+            index.insert(0, randomFrom("<", "-<"));
```

Comment:
This randomly generates excluding date math pattern

Reply:
Wouldn't it be better to randomly add a `minus` at the beginning, irrespective of date math? That's not related to date math, right? It's essentially another feature, like `EXCLUDE_PATTERN`.